explainer.md

```diff
@@ -111,34 +113,26 @@ code](demos/polyfill.js), which roughly implements this logic.
 ## LaunchEvent
 
     interface LaunchEvent : ExtendableEvent {
-      readonly attribute USVString url;
+      readonly attribute Request request;
       readonly attribute DOMString? clientId;
-
-      void handleLaunch(Promise<any> promise);
     }
 
-* `handleLaunch` is analogous to
+* `waitUntil` delays the user agent from launching and waits for the promise.
+  If the promise rejects, the default launch behaviour resumes.
+* `preventDefault` is analogous to
   [`FetchEvent`](https://www.w3.org/TR/service-workers-1/#fetch-event-section)'s
   `respondWith` method. If it is called (during the `launch` event handler), it
-  stops the user agent from launching and waits for the promise. If the promise
-  fulfills, nothing further happens (the user agent assumes the app has handled
-  it). If the promise rejects, the default launch behaviour resumes.
-* The `launch` event handler can also just call `preventDefault` to accomplish
-  the same thing (but this is more limited in that you can't decide
-  asynchronously whether to consume the event or not).
+  stops the user agent from launching. Nothing further happens (the user agent
+  assumes the app has handled it).
 * The `launch` event is considered to be "allowed to show a popup", so that
   `Clients.openWindow` and `Client.focus` can be used.
 
-Spec-ish text for `handleLaunch` algorithm with promise *promise* applied to
-event *event*:
-
-1. If internal slot [[*handlePending*]] is not null, throw `InvalidStateError`.
-2. Invoke *event*'s `waitUntil` with *promise*.
-3. Set internal slot [[*handlePending*]] to *promise*.
-
 ### Alternative design ideas and notes
 
-* `handleLaunch` could be given a promise that resolves to a Client or Client
+* We considered adding a `handleLaunch` method that accepts a promise like
```

Comment:
We should say why these ideas were rejected. I think `handleLaunch` is a totally different design where we return a disposition, as opposed to letting the declaratively handle it. Maybe we can explain that more here, so we don't end up circling back around on it one day.

Reply:
more detail added.